### PR TITLE
Removes some unnecessary dependencies on the public ts/ts.h

### PR DESCRIPTION
**Pull request by zwoop**
This also removes the proxy/api/ts directory from all default
search paths, to make it more difficult to introduce more bad
includes. There are still a few places that must have the public
APIs, and a few that does but shouldn't. This is a step right
direction at least.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1574/ for details.

**Comment by atsci** [comment]
Intel CC build *successful*! See https://ci.trafficserver.apache.org/job/icc-github/7/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1468/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/139/ for details.

File: lib/wccp/Makefile.am [hunk 1]
```diff
@@ -20,8 +20,8 @@
 include $(top_srcdir)/build/tidy.mk
 
 AM_CPPFLAGS = \
-  -I$(abs_top_srcdir)/lib \
-  -I$(abs_top_srcdir)/proxy/api/ts
+  -I$(abs_top_srcdir)/lib
```
Comment (jpeach) [inline lib/wccp/Makefile.am]: Missing a line continuation here.

File: proxy/FetchSM.cc [hunk 2]
```diff
@@ -26,6 +26,7 @@
 #include <stdio.h>
 #include "HTTP.h"
 #include "PluginVC.h"
+#include <api/ts/ts.h> // Ugly, but we need a bunch of the public APIs here ... :-/
```
Comment (jpeach) [inline proxy/FetchSM.cc]: Can you make this "api/ts/ts.h" for consistency with the other place you do this?

File: proxy/Main.cc [hunk 3]
```diff
@@ -38,6 +38,8 @@
 #include "ts/ink_syslog.h"
 #include "ts/hugepages.h"
 
+#include <api/ts/ts.h> // This is sadly needed becuase of us using TSThreadInit() for some reason.
```
Comment (jpeach) [inline proxy/Main.cc]: Consistency.

Comment (jpeach) [inline proxy/Main.cc]: s/becuase/because/

File: proxy/Plugin.cc [hunk 4]
```diff
@@ -231,8 +231,8 @@ plugin_init(bool validateOnly)
 
   if (INIT_ONCE) {
     api_init();
-    TSConfigDirGet();
-    plugin_dir = TSPluginDirGet();
+    RecConfigReadConfigDir(); // ToDo: Why is this needed? It does an initialization?
```
Comment (jpeach) [inline proxy/Plugin.cc]: `RecConfigReadConfigDir()` can't possibly be needed here.

File: proxy/http/remap/RemapConfig.cc [hunk 5]
```diff
@@ -778,7 +778,7 @@ remap_load_plugin(const char **argv, int argc, url_mapping *mp, char *errbuf, in
   }
 
   if (stat(c, &stat_buf) != 0) {
-    const char *plugin_default_path = TSPluginDirGet();
+    const char *plugin_default_path = RecConfigReadPrefixPath("proxy.config.plugin.plugin_dir");
```
Comment (jpeach) [inline proxy/http/remap/RemapConfig.cc]: You need to `free()` the result now.

File: proxy/Plugin.cc [hunk 6]
```diff
@@ -231,8 +231,8 @@ plugin_init(bool validateOnly)
...
+    plugin_dir = RecConfigReadPrefixPath("proxy.config.plugin.plugin_dir");
```
Comment (jpeach) [inline proxy/Plugin.cc]: You need to `free()` the result now.

Comment (zwoop) [inline proxy/Plugin.cc]: Hmmm, this one (plugin_dir) is saved as a global for eternity?

File: lib/wccp/Makefile.am (hunk 1 above)
Comment (zwoop) [inline lib/wccp/Makefile.am]: Wow, nice catch :)

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1495/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1601/ for details.

**Comment by atsci** [comment]
Intel CC build *successful*! See https://ci.trafficserver.apache.org/job/icc-github/33/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/165/ for details.

**Comment by zwoop** [comment]
@jablko Can you review please?

**Pull request merged**